pkg/hubble/relay/observer/server.go

```diff
@@ -168,6 +169,61 @@ sortedFlowsLoop:
 	return g.Wait()
 }
 
+// GetNodes implements observerpb.ObserverClient.GetNodes.
+func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest) (*observerpb.GetNodesResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+
+	peers := s.peers.List()
+	nodes := make([]*observerpb.Node, 0, len(peers))
+	for _, p := range s.peers.List() {
+		p := p
+		n := &observerpb.Node{
+			Name: p.Name,
+			Tls: &observerpb.TLS{
+				Enabled:    p.TLSEnabled,
+				ServerName: p.TLSServerName,
+			},
+		}
+		if p.Address != nil {
+			n.Address = p.Address.String()
+		}
+		nodes = append(nodes, n)
+		if !isAvailable(p.Conn) {
+			n.State = relaypb.NodeState_NODE_UNAVAILABLE
+			s.opts.log.WithField("address", p.Address).Infof(
+				"No connection to peer %s, skipping", p.Name,
+			)
+			s.peers.ReportOffline(p.Name)
+			continue
+		}
+		n.State = relaypb.NodeState_NODE_CONNECTED
+		g.Go(func() error {
+			n := n
+			client := s.opts.ocb.observerClient(&p)
+			status, err := client.ServerStatus(ctx, &observerpb.ServerStatusRequest{})
+			if err != nil {
+				n.State = relaypb.NodeState_NODE_ERROR
+				s.opts.log.WithFields(logrus.Fields{
+					"error": err,
+					"peer":  p,
+				}).Warning("Failed to retrieve server status")
+				return nil
```

Comment:
Question: maybe it depends on the error we get, but would it make sense to report offline here in some cases?

Reply:
That's a good point. Note that implementations of `ServerStatus` and `GetFlows` currently don't do this either.
I think this should be explored and eventually done as part of another PR.